src/base/dof_map.C

```diff
@@ -1483,7 +1483,13 @@ merge_ghost_functor_outputs(GhostingFunctor::map_type & elements_to_ghost,
                           temporary_coupling_matrices.find(const_cast<CouplingMatrix *>(existing_it->second)) == temporary_coupling_matrices.end())
                         {
                           CouplingMatrix * cm = new CouplingMatrix(*existing_it->second);
-                          temporary_coupling_matrices.insert(cm);
+#ifndef NDEBUG
+                          auto dbg_pr =
+#endif
+                            temporary_coupling_matrices.insert(cm);
+                          libmesh_assert_msg(dbg_pr.second,
+                                             "We must actually insert this new coupling matrix or "
+                                             "else we are going to leak");
```

Comment:
How is this ever going to fail?  If we just got a pointer from new, then it won't alias any existing pointer and it won't be in the container.  If we didn't, then new should have thrown an exception and we shouldn't have made it to the assert.

Reply:
Yea...that seems pretty obvious when I give any level of thought about it